fix TestPausedProvisionerProvision_waits, that randomly fails

This test is flaky.

```
--- FAIL: TestPausedProvisionerProvision_waits (0.14s)
    provisioner_test.go:151: never called
2019/05/24 14:42:02 ui: Pausing 0s before the next provisioner...
2019/05/24 14:42:02 ui: ask: Pausing before the next provisioner . Press enter to continue.
2019/05/24 14:42:02 ui: ask: Pausing before the next provisioner . Press enter to continue.
2019/05/24 14:42:02 ui: foo
2019/05/24 14:42:02 ui: foo
2019/05/24 14:42:02 ui error: foo
2019/05/24 14:42:02 ui: foo
2019/05/24 14:42:02 ui: foo
2019/05/24 14:42:02 ui error: foo
2019/05/24 14:42:02 ui: ==> foo: foo
2019/05/24 14:42:02 ui:     foo: foo
2019/05/24 14:42:02 ui error: ==> foo: bar
2019/05/24 14:42:02 ui: ==> foo: foo
==> foo: bar
2019/05/24 14:42:02 ui error: foo
2019/05/24 14:42:02 ui error: 5
2019/05/24 14:42:02 ui: foo
2019/05/24 14:42:02 ui: 5
2019/05/24 14:42:02 ui: ask: [c]ontinue or [a]bort
2019/05/24 14:42:02 ui: ask: [c]ontinue or [a]bort
2019/05/24 14:42:02 ui: ask: Name
2019/05/24 14:42:02 ui: ask: Name
2019/05/24 14:42:02 ui: ask: Name
```

Ideally the provisioner should just contain the code that checks time spent.

https://github.com/hashicorp/packer/blob/82c4b766398a6383ff28c3d85e722e06abd3dd05/packer/provisioner_test.go#L128-L154

--- a/packer/provisioner_test.go
+++ b/packer/provisioner_test.go
@@ -138,9 +138,6 @@ func TestPausedProvisionerProvision_waits(t *testing.T) {
 				if timeSinceStartTime < waitTime {
 					return fmt.Errorf("Spent not enough time waiting: %s", timeSinceStartTime)
 				}
-				if timeSinceStartTime > waitTime+10*time.Millisecond {
-					return fmt.Errorf("Spent too much time waiting: %s", timeSinceStartTime)
-				}
 				return nil
 			},
 		},

--- a/packer/provisioner_test.go
+++ b/packer/provisioner_test.go
@@ -2,6 +2,7 @@ package packer
 
 import (
 	"context"
+	"fmt"
 	"testing"
 	"time"
 )
@@ -126,30 +127,29 @@ func TestPausedProvisionerProvision(t *testing.T) {
 }
 
 func TestPausedProvisionerProvision_waits(t *testing.T) {
-	mock := new(MockProvisioner)
+	startTime := time.Now()
+	waitTime := 50 * time.Millisecond
+
 	prov := &PausedProvisioner{
-		PauseBefore: 50 * time.Millisecond,
-		Provisioner: mock,
+		PauseBefore: waitTime,
+		Provisioner: &MockProvisioner{
+			ProvFunc: func(context.Context) error {
+				timeSinceStartTime := time.Since(startTime)
+				if timeSinceStartTime < waitTime {
+					return fmt.Errorf("Spent not enough time waiting: %s", timeSinceStartTime)
+				}
+				if timeSinceStartTime > waitTime+10*time.Millisecond {
+					return fmt.Errorf("Spent too much time waiting: %s", timeSinceStartTime)
+				}
+				return nil
+			},
+		},
 	}
 
-	dataCh := make(chan struct{})
-	mock.ProvFunc = func(context.Context) error {
-		close(dataCh)
-		return nil
-	}
+	err := prov.Provision(context.Background(), testUi(), new(MockCommunicator))
 
-	go prov.Provision(context.Background(), testUi(), new(MockCommunicator))
-
-	select {
-	case <-time.After(10 * time.Millisecond):
-	case <-dataCh:
-		t.Fatal("should not be called")
-	}
-
-	select {
-	case <-time.After(100 * time.Millisecond):
-		t.Fatal("never called")
-	case <-dataCh:
+	if err != nil {
+		t.Fatalf("prov failed: %v", err)
 	}
 }
 

